examples/qtmegachatapi/MegaChatApplication.cpp

```diff
@@ -680,21 +680,31 @@ void MegaChatApplication::onRequestFinish(MegaChatApi *, MegaChatRequest *reques
              {
              MegaChatHandle userHandle = request->getUserHandle();
              int errorCode = error;
+             mFirstnameFetching[userHandle] = false;
              if (errorCode == MegaChatError::ERROR_OK)
              {
                 const char *firstname = request->getText();
                 if ((strlen(firstname)) == 0)
                 {
-                    this->mMegaChatApi->getUserEmail(userHandle);
+                    auto it = mFirstnamesMap.find(userHandle);
+                    if (it != mFirstnamesMap.end())
+                    {
+                        mFirstnamesMap.erase(it);
+                    }
+                    mMegaChatApi->getUserEmail(userHandle);
                     break;
                 }
                 mFirstnamesMap[userHandle] = firstname;
-                mFirstnameFetching[userHandle] = false;
                 mMainWin->updateContactTitle(userHandle,firstname);
                 mMainWin->updateMessageFirstname(userHandle,firstname);
              }
              else if (errorCode == MegaChatError::ERROR_NOENT)
              {
+                auto it = mFirstnamesMap.find(userHandle);
```

Comment:
It's not necessary `find` in a map before `erase`. `erase` from a map with a non-existent key is safe. The code is correct. You can remove if you want or keep in this way